2016-04/april-21.md

```diff
@@ -0,0 +1,79 @@
+
+### 15.0 Post Mortem
+
+#### Release was pretty rocky
+* We didn’t have a cohesive enough story around what was to be included in the release.
+* Now that we have more dependants (RN, many teams internally and externally) it’s becoming an issue.
+* Examples of changes that should we should have planned better:
+  * [SVG attributes were passed through without a whitelist in RC1](https://github.com/facebook/react/pull/5714), then [we reverted this in RC2](https://github.com/zpao/react/commit/08fa7fe50707d4c7fe06861e0875b3fab00ea048).
+  * We [switched to using attributes](https://github.com/facebook/react/pull/1510) by default but haven’t tested this as thoroughly as we should have, and it caused a few regressions, plus later it turned out that we didn’t all agree on this change ([Sebastian](https://twitter.com/sebmarkbage) was out of the loop).
+* Our default in the past has been to just work on everything individually.
+  * [Ben](https://twitter.com/soprano) created a special GitHub group so that we can auto-subscribe all of us to some big picture issues.
+* We should have shipped an RC3.
+  * But why did we get to RC2?
+  * Having browser tests ([Jim](https://github.com/jimfb)) and planning releases ([Paul](https://twitter.com/zpao)) more thoroughly should fix this.
+* We should make the release process easier.
+  * Right now there’s too much manual work involved.
+  * If it was just one command, maybe we would have been more open to releasing RC3.
+
+### Releasing 15.0.2
+
+* Either this Friday or early next week.
+* We merged some changes to [bring the `react-native-renderer` into the React repo](https://github.com/facebook/react/pull/6338) so we can change React internals more freely without breaking RN.
+* What else should get into 15.0.2?
+  * [Fix for the `optgroup`s](https://github.com/facebook/react/pull/6442).
+  * [CSS warning fix](https://github.com/facebook/react/pull/6458).
+  * [TestUtils fix](https://github.com/facebook/react/pull/6362).
+* Right now we just run master on Facebook.
+* In six months, master and 15.x can diverge significantly.
+* We need to figure out a better story around syncing React with:
+  * The Facebook website.
+  * React Native.
+* [Paul](https://twitter.com/zpao) will write up how we’re going to do releases moving forward.
+
+### Universal Components
+
+* [Nicolas](https://github.com/necolas) from Twitter created [React Native Web](https://github.com/necolas/react-native-web).
+* [Leland](https://github.com/lelandrichardson) from Airbnb maintains [a fork](https://github.com/lelandrichardson/react-native-web).
+* It is not clear how to create universal components.
+* For example, if you depend on `View` from `react-native`, you pull in the whole `react-native`.
+* How do you create a component that is renderable both in React Native and React Native Web?
+  * Some registration system?
+  * Injectable components?
+  * Strings?
+* No conclusion, lots to think about.
+
+### Adding Flow to React
```

Comment:
I'm not sure this the best place for such discussions, but have you considered typescript as an alternative for flow? Would be great to hear team's opinion about it.